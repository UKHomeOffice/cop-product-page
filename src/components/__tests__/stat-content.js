import React from "react";
import renderer from "react-test-renderer";
import Header from "../header";
import StatContent from "../stats-content";

describe("stat-content", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<StatContent title={"Used and trusted by Border Force teams"}
                           stats={[{ number: "6,652", description: "Border Force users" },
                             { number: "7,500+", description: "forms and reports submitted per month" },
                             { number: "300+", description: "COP visits per day" }]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays the service name", () => {
    const tree = renderer
      .create(<StatContent title={"Used and trusted by Border Force teams"}
                           stats={[{ number: "6,652", description: "Border Force users" },
                             { number: "7,500+", description: "forms and reports submitted per month" },
                             { number: "300+", description: "COP visits per day" }]}/>);
    const testInstance = tree.root;
    const getProp = (key) => {
      return testInstance.findByProps({ id: key })._fiber.memoizedProps.text;
    };
    expect(getProp("valueText0")).toEqual("Border Force users");
    expect(getProp("valueText1")).toEqual("forms and reports submitted per month");
    expect(getProp("valueText2")).toEqual("COP visits per day");
    expect(getProp("valueNum0")).toEqual("6,652");
    expect(getProp("valueNum1")).toEqual("7,500+");
    expect(getProp("valueNum2")).toEqual("300+");
  });
});