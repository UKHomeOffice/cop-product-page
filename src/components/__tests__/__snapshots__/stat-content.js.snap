// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stat-content renders correctly 1`] = `
<div>
  <h3
    className="govuk-heading-m"
    id="title"
  >
    Used and trusted by Border Force teams
  </h3>
  <div
    style={
      Object {
        "marginTop": "50px",
      }
    }
  />
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-one-third"
    >
      <h1
        className="govuk-heading-xl"
      >
        6,6525
      </h1>
      <p
        className="govuk-body"
        id="valueText0"
        style={
          Object {
            "maxWidth": "75%",
          }
        }
      >
        Border Force users
      </p>
    </div>
    <div
      className="govuk-grid-column-one-third"
    >
      <h1
        className="govuk-heading-xl"
      >
        7,55500+
      </h1>
      <p
        className="govuk-body"
        id="valueText1"
        style={
          Object {
            "maxWidth": "75%",
          }
        }
      >
        forms and reports submitted per month
      </p>
    </div>
    <div
      className="govuk-grid-column-one-third"
    >
      <h1
        className="govuk-heading-xl"
      >
        33300+
      </h1>
      <p
        className="govuk-body"
        id="valueText2"
        style={
          Object {
            "maxWidth": "75%",
          }
        }
      >
        COP visits per day
      </p>
    </div>
  </div>
</div>
`;
